### ISSUE TITLE
Avoid panic for invalid port config

### ISSUE THREAD

**Issue by nikkolasg** [drand/drand#675]
For example drand panics when your private listen and control port are the same

**Comment by AnomalRoil** [comment]
It will also panic when using a control port that is already used (e.g. by another node).
We should error out and exit gracefully with a useful error message instead.

**Comment by AnomalRoil** [comment]
Fixed in #1030:
```
$ ./drand start --tls-disable --folder ./local-tests/b2 --private-listen 127.0.0.1:7002 --control 7002 -verbose
drand 1.4.6-testnet (date 07/09/2022@20:09:45, commit a2a2257)
2022-09-13T16:11:23.048+0200	INFO	127.0.0.1:7002	core/drand_daemon.go:115		{"network": "init", "insecure": true}
2022-09-13T16:11:23.049+0200	ERROR	net/control.go:32		{"grpc listener": "failure", "err": "listen tcp 127.0.0.1:7002: bind: address already in use"}
can't instantiate drand daemon listen tcp 127.0.0.1:7002: bind: address already in use
```
